java/client/src/main/java/glide/api/models/commands/PasswordUpdateMode.java

```diff
@@ -0,0 +1,19 @@
+/** Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.api.models.commands;
+
+public enum PasswordUpdateMode {
+    /** The client will re-authenticate immediately with the new password. */
+    RE_AUTHENTICATE(true),
+    /** The new password will be used for the next connection attempt. */
+    USE_ON_NEW_CONNECTION(false);
```

Comment:
What does it mean? The whole meaning is that it will be used on a new connection attempt, there's no other scenario the password will be used.
If the goal was to simply perform re-auth, we could just send `HELLO pass`, `PING pass` or simply use `AUTH pass`.
The re-auth is a nice to have, that's it.

Reply:
Maybe it wasnt clear, the reauth is to perform or not reauth. if the reauth not used, or if it does, the password will be used on a new connection. The point mentioned in the doc were about the fact that if you dont perform re-auth, it will be used just on the next reconnection attempt, and nothing else until then.